### PR TITLE
Implement reformat of tuple litterals

### PR DESCRIPTION
**Pull request by cassiersg**

### CODE REVIEW

File: src/expr.rs [hunk 1]
```diff
@@ -194,6 +194,40 @@ impl<'a> FmtVisitor<'a> {
...
+        -> String {
+        // opening paren
+        let indent = offset + 1;
+        // Only last line has width-1 as budget, other may take max_width
+        let item_strs: Vec<_> =
+            items.iter()
+                 .enumerate()
+                 .map(|(i, item)| self.rewrite_expr(
+                    item,
+                    // for last line, -2 is for indent + ")", for other lines, -1 is for comma
```
Comment (nrc) [inline src/expr.rs]: I think the 'other lines' should also be -2 because they are also offset by the opening `(`

**Comment by nrc** [comment]
Do you handle tuples with len == 1 properly? These should be formatted as `(foo,)` rather than `(foo)`. This might require a new SeperatorTactic. It'll need a test in any case.

Otherwise, this looks good!

File: src/expr.rs (hunk 1 above)
Comment (cassiersg) [inline src/expr.rs]: Indeed.

**Comment by cassiersg** [comment]
I simply added a special case for tuples of length 1, the general algo is not adapted because `width - 3` is needed to get comma and paren on same line.

**Pull request merged**